src/mpyl/steps/deploy/k8s/cluster.py

```diff
@@ -0,0 +1,85 @@
+""" Utilities for creating rancher compatible helm charts. """
+
+from dataclasses import dataclass
+from typing import Optional, Any
+
+from ...models import RunProperties
+from ....project import TargetProperty, Project
+
+
+@dataclass(frozen=True)
+class ClusterConfig:
+    name: str
+    cluster_id: Optional[str]
+    cluster_env: str
+    context: str
+
+    @staticmethod
+    def from_config(config: dict):
+        return ClusterConfig(
+            name=config["name"],
+            cluster_id=config.get("clusterId"),
+            cluster_env=config["clusterEnv"],
+            context=config["context"],
+        )
+
+
+def get_cluster_config_for_project(
+    run_properties: RunProperties, project: Project
+) -> ClusterConfig:
+    cluster_override_name = (
+        project.deployment.cluster.get_value(run_properties.target)
+        if project.deployment and project.deployment.cluster
+        else None
+    )
+
+    kubernetes_config = run_properties.config["kubernetes"]
+
+    clusters = [
+        ClusterConfig.from_config(cluster_config)
+        for cluster_config in kubernetes_config["clusters"]
+    ]
+
+    default_cluster_name: str = TargetProperty.from_config(
+        kubernetes_config["defaultCluster"]
+    ).get_value(run_properties.target)
+
+    default_cluster: ClusterConfig = next(
```

Comment:
I would maybe extract the logic to get the default cluster to its own function and only call it in the `else` on line 65. Makes this whole thing easier to read about what's happening

Reply:
done in [d297458](https://github.com/Vandebron/mpyl/pull/335/commits/d2974582b15707f23620907591b70514a1524a36)

moved it around, but didn't do exactly like you said, so we can still keep the exception if the default is not found